modules/nft/keeper/keeper.go

```diff
@@ -47,6 +49,11 @@ func (k Keeper) MintNFT(
 		return sdkerrors.Wrapf(types.ErrInvalidDenom, "denom ID %s not exists", denomID)
 	}
 
+	denom, _ := k.GetDenom(ctx, denomID)
```

Comment:
```suggestion
	denom, found := k.GetDenom(ctx, denomID)
	if !found{
	    return sdkerrors.Wrapf(types.ErrInvalidDenom, "denom ID %s not exists", denomID)
	}
```
remove `k.HasDenomID(ctx, denomID)`